### PR TITLE
fix: Repetition of root_path in supertokens mididdlware for fastapi

### PR DESCRIPTION
**Pull request by KShivendu**
## Summary of change

Fix repetition of root_path in supertokens mididdlware for fastapi

## Related issues

-   https://github.com/supertokens/supertokens-python/issues/230

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py

### CODE REVIEW

File: supertokens_python/framework/fastapi/fastapi_request.py [hunk 1]
```diff
@@ -60,7 +60,9 @@ def set_session_as_none(self):
         self.request.state.supertokens = None
 
     def get_path(self) -> str:
-        return self.request.url.path
+        root_path = self.request.scope.get("root_path")
+        url = self.request.url.path
+        return url[url.startswith(root_path) and len(root_path) :]
```
Comment (rishabhpoddar) [inline supertokens_python/framework/fastapi/fastapi_request.py:65]: - add tests for this?
- add comments exaplaining this? 
- What are all the edge cases this covers?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/framework/fastapi/fastapi_request.py [hunk 2]
```diff
@@ -60,7 +60,11 @@ def set_session_as_none(self):
         self.request.state.supertokens = None
 
     def get_path(self) -> str:
-        return self.request.url.path
+        root_path = self.request.scope.get("root_path")
+        url = self.request.url.path
+        # FastAPI seems buggy and it adds an extra root_path (if it matches):
+        # So we trim the extra root_path (from the left) from the url
```
Comment (rishabhpoddar) [inline supertokens_python/framework/fastapi/fastapi_request.py:66]: - "if it matches" -> matches what?
- what is the extra root path doesn't match? Then what will happen?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]
